14/README.md

```diff
@@ -0,0 +1,173 @@
+
+stream-like: ```DEBUG_STREAM << "Msg#" << i++ << "- Hello world" << endl;```
+
+These two logging requests are equivalent. Note the double parenthesis in the printf version.
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+TLS SHALL provide a means to log in the name of a device.  This SHALL be provided when the Device must is a member of a class inheriting from the ```Tango::LogAdapter``` class.
+
+### Python API
+TLS SHALL provide the folliwng Python forms of writing log messages:
+
+basic logging: ```self.debug_stream("read_voltage(%s, %d)", self.host, self.port)```
+
+logging with print: ```print >>self.debug_info, "read voltage attribute"```
+
+logging with decorators: ```@PyTango.DebugIt()```
+
+The examples just given are for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+
+### Java API
+TLS SHALL provide configuration to allow Java to log messages in the following manner:   
+```get_logger().debug(Initializing device  + get_name());```   
+The example just given is for the DEBUG log level.  There SHALL BE equivalent macros for the other log levels, except OFF.
+
+
+### Log Consumer
+When a Device Target is specified as "device", this SHALL designate a Device Server which is implementing the Log Consumer interface.
+To be a Log Consumer, a Tango Device SHALL implement just one Tango Command:   
+```void log (Tango::DevVarStringArray details)```.   
+
+The array of strings composing the argument SHALL be interpreted with the following meanings:
+- details[0] : the timestamp in millisecond since epoch (01.01.1970)
+- details[1] : the log level
+- details[2] : the log source (i.e. device name)
+- details[3] : the log message
+- details[4] : the log NDC (contextual info) - Not used but reserved
+- details[5] : the thread identifier (i.e. the thread from which the log request comes from)
+
+
+### GUI for viewing logs
+TLS SHALL include a GUI application called LogViewer or displaying logs pushed by the devices to the Tango logging system. 
+LogViewer SHALL have the ability to fetch logs from different Tango devices and filter them by several properties.
+The LogViewer SHALL have the following abilities:
+- show logging messages from multiple devices simultaneously
+- filter logging messages by Logging Level
+- be able to read log files from disk
+
+LogViewer SHALL have two modes: static and dynamic.   
+
+- static: LogViewer SHALL be started with the name of the Log Consumer Device name and it SHALL receive all messages from Devices who have this name as their target.
+
+- dynamic: LogViewer SHALL let the user use the GUI to specify which devices LogViewer will monitor.
+
+
+## References
+The TANGO Control System: Reference Section A.1.5: The device logging
+http://www.esrf.eu/computing/cs/tango/tango_doc/kernel_doc/ds_prog/node11.html#SECTION001115000000000000000
```

Comment:
All links must point to https://tango-controls.readthedocs.io i.e. official Tango documentation not ESRF

Reply:
Agreed, and done.